nixos/tests/installer.nix

```diff
@@ -685,6 +685,65 @@ in {
     '';
   };
 
+  bcachefsSimple = makeInstallerTest "bcachefs-simple" {
+    extraInstallerConfig = {
+      boot.supportedFilesystems = [ "bcachefs" ];
+    };
+
+    extraConfig = ''
+      boot.supportedFilesystems = [ "bcachefs" ];
+    '';
+
+    createPartitions = ''
+      machine.succeed(
+          "sgdisk -Z /dev/vda",
+          "sgdisk -n 1:0:+1M -N 2 -t 1:ef02 -t 2:8304 -c 2:root /dev/vda",
+          "mkfs.bcachefs -L root /dev/vda2",
+          "mount -t bcachefs /dev/vda2 /mnt",
+      )
+    '';
+  };
+
+  # Due to #32279, we cannot use this test yet
+#  bcachefsEncrypted = makeInstallerTest "bcachefs-encrypted" {
+#    extraInstallerConfig = {
+#      boot.supportedFilesystems = [ "bcachefs" ];
+#    };
+
+#    extraConfig = ''
+#      boot.supportedFilesystems = [ "bcachefs" ];
+#    '';
+
+#    createPartitions = ''
+#      machine.succeed(
+#          "sgdisk -Z /dev/vda",
+#          "sgdisk -n 1:0:+1M -N 2 -t 1:ef02 -t 2:8304 -c 2:root /dev/vda",
+#          "echo password | mkfs.bcachefs -L root --encrypted /dev/vda2",
+#          "echo password | bcachefs unlock /dev/vda2",
+#          "mount -t bcachefs /dev/vda2 /mnt",
+#      )
+#    '';
+#  };
+
+  bcachefsMulti = makeInstallerTest "bcachefs-multi" {
+    extraInstallerConfig = {
+      boot.supportedFilesystems = [ "bcachefs" ];
+    };
+
+    extraConfig = ''
+      boot.supportedFilesystems = [ "bcachefs" ];
+    '';
+
+    createPartitions = ''
+      machine.succeed(
+          "sgdisk -Z /dev/vda",
+          "sgdisk -n 1:0:+1M -n 2:0:+100M -n 3:0:+200M -N 4 -t 1:ef02 -t 2:8304 -t 3:8304 -t 4:8304 -c 2:root -c 3:root -c 4:root /dev/vda",
+          "mkfs.bcachefs -L root --metadata_replicas 2 /dev/vda2 /dev/vda3 /dev/vda4",
+          "mount -t bcachefs /dev/vda2:/dev/vda3:/dev/vda4 /mnt",
```

Comment:
I notice that the front page of https://bcachefs.org/ gives this as the standard 'getting started' multi-device bcache filesystem:

```
bcachefs format /dev/sd[ab]1 \
    --foreground_target /dev/sda1 \
    --promote-target /dev/sda1 \
    --background_target /dev/sdb1
```

Ideally, our test would be as much like this as possible.

[When I try this](https://github.com/chkno/nixpkgs/commit/2c1045fcbce0e3fda34d92691886d21ee15562f4), it doesn't work. I get 

```
bcachefs: unrecognized option '--promote-target'
```
It seems to me that if users can't actually do what bcachefs.org says to do, NixOS doesn't really support bcachefs yet.

Can we make this canonical bcachefs.org invocation work, and then use it here for this test to ensure that it continues to work?

Reply:
bcachefs.org is outdated IIRC, also that seems to be a typo on their part, should be ``--promote_target``.
This is what should be used. https://bcachefs.org/bcachefs-principles-of-operation.pdf
Anyway yeah, I'll add the targets.

Reply:
Ah, ok.

Looks like the bcachefs wiki is edited by cloning https://evilpiepirate.org/git/bcachefs-wiki.git and sending patches to the "linux-bcache" vger.kernel.org mailing list (linked from https://bcachefs.org/Contact/ ).